Sources/Sentry/SentryScope.m

```diff
@@ -330,6 +364,13 @@ - (void)addAttachment:(SentryAttachment *)attachment
     [serializedData setValue:self.environmentString forKey:@"environment"];
     [serializedData setValue:[self fingerprints] forKey:@"fingerprint"];
 
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]])
+        [serializedData setValue:[(SentryTracer *)self.currentTransaction name]
+                          forKey:@"transaction"];
+    else if (self.fallbackTransactionName != nil)
+        [serializedData setValue:self.fallbackTransactionName forKey:@"transaction"];
```

Comment:
All of this is racy.
Remember that SentryTracer was designed to be access concurrently. It will be getting calls to `startSpan` potentially while you're now trying to serialize it.

There's a lot of synchronization work needed here. One approach taken by @philipphofmann in the past was to have a clone method on the instance that would synchronize access to its field and return a new instance (clone) that could be used safely for serialization.